tracing-core/src/dispatcher.rs

```diff
@@ -711,11 +731,11 @@ impl State {
 #[cfg(feature = "std")]
 impl<'a> Entered<'a> {
     #[inline]
-    fn current(&self) -> RefMut<'a, Dispatch> {
-        let default = self.0.default.borrow_mut();
-        RefMut::map(default, |default| {
-            default.get_or_insert_with(|| get_global().cloned().unwrap_or_else(Dispatch::none))
-        })
+    fn with_current<T>(&self, f: impl FnOnce(&Dispatch) -> T) -> T {
+        match &*self.0.default.borrow() {
+            Some(default) => f(default),
+            None => unsafe { f(&*GLOBAL_DEFAULT_DISPATCH.load(Ordering::SeqCst)) },
```

Comment:
are we sure this load needs to be sequentially consistent, or can it just be an `Acquire` load? what other atomic value does this need to synchronize with?